processing/src/main/java/org/apache/druid/query/InlineDataSource.java

```diff
@@ -252,7 +252,7 @@ public DataSource withUpdatedDataSource(DataSource newSource)
   @Override
   public byte[] getCacheKey()
   {
-    return null;
+    return new byte[0];
```

Comment:
Are you sure that this was cached previously?  It seems like an `InlineDataSource` would either have to include all of the data in the cache key or cause caching to never happen...